src/RoadRegistry.Syndication.Projections/MunicipalityCacheProjection.cs

```diff
@@ -0,0 +1,183 @@
+namespace RoadRegistry.Syndication.Projections
+{
+    using System;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.Connector;
+    using Be.Vlaanderen.Basisregisters.ProjectionHandling.SqlStreamStore;
+    using MunicipalityEvents;
+    using Schema;
+
+    public class MunicipalityCacheProjection : ConnectedProjection<SyndicationContext>
+    {
+        public MunicipalityCacheProjection()
+        {
+            When<Envelope<MunicipalityWasRegistered>>(async (context, envelope, token) =>
+            {
+                await context.Municipalities.AddAsync(
+                    new MunicipalityRecord
+                    {
+                        MunicipalityId = envelope.Message.MunicipalityId,
+                        NisCode = envelope.Message.NisCode,
+                        DutchName = null,
+                        FrenchName = null,
+                        GermanName = null,
+                        EnglishName = null,
+                        MunicipalityStatus = MunicipalityStatus.Registered
+                    }, token);
+            });
+
+            When<Envelope<MunicipalityWasNamed>>(async (context, envelope, token) =>
+            {
+                var municipalityRecord = await context.Municipalities.FindAsync(envelope.Message.MunicipalityId);
+                if (municipalityRecord == null)
```

Comment:
Not sure I like this, I'd rather have it blow up to be honest, this is just sweeping a problem under the carpet only to have us go on a wild goose chase as to why the record is not there. Let's fix the underlying problem, not cascade it.